corehq/apps/geospatial/routing_solvers/pulp.py

```diff
@@ -33,52 +33,94 @@ def calculate_distance_matrix(self, config):
+        # Add constraints
+        problem = self.add_user_case_assignment_constraint(
+            lp_problem=problem,
+            decision_variables=decision_variables,
+            user_count=user_count,
+            case_count=case_count,
+            min_cases=config.min_cases_per_user,
+            max_cases=config.max_cases_per_user,
+        )
+        problem = self.add_case_owner_constraint(
+            lp_problem=problem,
+            decision_variables=decision_variables,
+            user_count=user_count,
+            case_count=case_count,
+        )
 
         # Define the objective function
-        objective_terms = [costs[i][j] * x[i, j] for i in range(user_count) for j in range(case_count)]
+        objective_terms = [
+            costs[i][j] * decision_variables[i, j]
+            for i in range(user_count) for j in range(case_count)
+        ]
         problem += pulp.lpSum(objective_terms)
 
         # Solve the problem
         problem.solve()
 
-        solution = {loc['id']: [] for loc in self.user_locations}
-
         # Process the solution
         if pulp.LpStatus[problem.status] == "Optimal":
-            if print_solution:
-                print(f"Total cost = {pulp.value(problem.objective)}\n")
+            solution = {loc['id']: [] for loc in self.user_locations}
+
             for i in range(user_count):
                 for j in range(case_count):
-                    if pulp.value(x[i, j]) > 0.5:
+                    if pulp.value(decision_variables[i, j]) > 0.5:
                         solution[self.user_locations[i]['id']].append(self.case_locations[j]['id'])
-                        if print_solution:
-                            print(f"Case {self.case_locations[j]['id']} assigned to "
-                                  f"user {self.user_locations[i]['id']}. "
-                                  f"Cost: {costs[i][j]}")
-        else:
-            if print_solution:
-                print("No solution found.")
-        return None, solution
+            return None, solution
+
+        return None, None
+
+    @staticmethod
+    def get_decision_variables(x_dim, y_dim):
+        matrix = {}
+        for i in range(x_dim):
+            for j in range(y_dim):
+                matrix[i, j] = pulp.LpVariable(f"x_{i}_{j}", lowBound=0, upBound=1, cat=pulp.LpBinary)
+        return matrix
+
+    @staticmethod
+    def add_user_case_assignment_constraint(
+        lp_problem, decision_variables, user_count, case_count, min_cases=None, max_cases=None
+    ):
+        # This constrain enforces the min/max amount of cases that could be assigned to each user,
+        # with the default being the cases split equally between users.
+        max_constraint = int(case_count / user_count) + 1
+        min_constraint = 0
+
+        if min_cases:
+            min_constraint = min_cases
+        if max_cases:
+            max_constraint = max_cases
```

Comment:
nit: would be nice to make this explicit. It's more of a personal preference. I believe it is confusing when there is a default value set that is changed later in the function.

```
min_constraint = min_cases or 0
max_constraint = max_cases or int(case_count / user_count) + 1
```